docs/DeveloperGuide.adoc

```diff
@@ -571,24 +571,68 @@ image::implementation/BudgetPanelSequenceDiagram.png[width="800"]
 
 === Budget
 ==== Current Implementation
+The budget feature is one whereby after a user sets a budget, if the total spendings of a user exceeds his budget
```

Comment:
This sentence is a bit too long. Maybe can rephrase to something like...

The budget feature allows the user to set a budget for a specific time period. If the user's total spending exceeds their budget in the specified time period, a warning will be shown to the user.